include/openspace/scene/asset.h

```diff
@@ -26,156 +26,277 @@
 #define __OPENSPACE_CORE___ASSET___H__
 
 #include <openspace/util/resourcesynchronization.h>
-#include <openspace/util/synchronizationwatcher.h>
-#include <memory>
+#include <filesystem>
 #include <optional>
-#include <string>
-#include <vector>
 
 namespace openspace {
 
-class AssetLoader;
-
-class Asset : public std::enable_shared_from_this<Asset> {
+class AssetManager;
+
+/**
+ * This class represents a successfully loaded Asset. Each Lua asset file results in
+ * exactly one instance of this class unless it contains a syntax error or some other
+ * error occurred while loading. Each Asset can have 0-* number of resource
+ * synchronizations that were requested through the `syncedResource` function in Lua and
+ * 0-* number of other Assets that this Asset requires (through the `require` function in
+ * Lua. There also is a list of requiring assets that contain all assets which require
+ * this asset.
+ * An asset goes through three external states. Starting out as unloaded when the instance
+ * is newly created but the file has not been processed. In this case the #isLoaded,
+ * #isSynchronized and the #isInitialized functions all return \c false. After the asset
+ * has been loaded it is in the \c Loaded state (#isLoaded = true,
+ * #isSynchronized = false, #isInitialized = false). After all registered synchronizations
+ * finish successfully, the Asset transitions into the Synchronized state
+ * (#isLoaded = true, #isSynchronized = true, #isInitialized = false) and after the final
+ * initialization step, the asset is initialized (#isLoaded = true,
+ * #isSynchronized = true and #isInitialized = true)
+ */
+class Asset {
 public:
-    enum class State {
-        Unloaded,
-        LoadingFailed,
-        Loaded,
-        Synchronizing,
-        SyncResolved,
-        SyncRejected,
-        Initialized,
-        InitializationFailed
-    };
-
+    /// This struct contains all the meta information about this asset
     struct MetaInformation {
+        /// The name of the asset
```

Comment:
Just a question about this comment style, with three `/`: What are the situations where we prefer this one over just two slashes?

Reply:
I don't think we have thought about it.  The way I'm using it is either
 1. It's a one line comment anywhere
 2. It's a one or two line comment in the `private` section of a class

But happy to talk about it and come to a consensus